r/R/util.R

```diff
@@ -110,3 +110,21 @@ handle_embedded_nul_error <- function(e) {
   }
   stop(e)
 }
+
+#' Take an object of length 1 and repeat it.
+#' 
+#' @param object Object to be repeated - vector, `Scalar`, `Array`, or `ChunkedArray`
+#' @param n Number of repetitions
+#' 
+#' @return `Array` of length `n`
+#' 
+#' @keywords internal
+repeat_value_as_array <- function(object, n) {
+  if (length(object) != 1) {
```

Comment:
If you're checking length again here (unnecessary IMO since you're only calling this in the case where you've validated length == 1 already), you could simplify your `modify2()` wrapper and just `map` over all `arrays`, and in here only do the recycling if length == 1.

Reply:
I think I was trying to make this function a bit more generic in case it's useful elsewhere, but you make a good point; I'll remove the check.